Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -150,9 +154,43 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           functionSymbol.Parameters.Add(ProcessFormal(scope, parameter));
         }
+        if (function.Result != null) {
+          functionSymbol.Parameters.Add(ProcessFormal(scope, function.Result));
+        }
+        ScopeSymbol? ExpressionHandler(Expression? expression) =>
+          expression == null
+            ? null
+            : ProcessExpression(new ScopeSymbol(functionSymbol, functionSymbol.Declaration), expression);
+        functionSymbol.Body = ExpressionHandler(function.Body);
+        functionSymbol.Ensures.AddRange(ProcessListAttributedExpressions(function.Ens, ExpressionHandler));
+        functionSymbol.Requires.AddRange(ProcessListAttributedExpressions(function.Req, ExpressionHandler));
+        functionSymbol.Reads.AddRange(ProcessListFramedExpressions(function.Reads, ExpressionHandler));
+        functionSymbol.Decreases.AddRange(ProcessListExpressions(function.Decreases.Expressions, ExpressionHandler));
+        functionSymbol.ByMethodBody = function.ByMethodBody == null ? null : ProcessFunctionByMethodBody(functionSymbol, function.ByMethodBody);
         return functionSymbol;
       }
 
+      private List<ScopeSymbol> ProcessListAttributedExpressions(
+        IList<AttributedExpression> list, Func<Expression?, ScopeSymbol?> expressionHandler
+        ) {
+        return list.SelectMany(attributedExpression =>
+          expressionHandler(attributedExpression.E).AsEnumerable()).ToList();
+      }
+
+      private List<ScopeSymbol> ProcessListFramedExpressions(
+        IList<FrameExpression> list, Func<Expression?, ScopeSymbol?> expressionHandler
+        ) {
+        return list.SelectMany(frameExpression =>
+          expressionHandler(frameExpression.E).AsEnumerable()).ToList();
+      }
+
+      private List<ScopeSymbol> ProcessListExpressions<T>(
+        List<T> list, Func<T, ScopeSymbol?> expressionHandler
```

Comment:
`List<T>` appears to be unnecessary. `IEnumerable<T>` should be enough. This saves unnecessary `ToList()` calls.
Moreover, the final `ToList()` is probably unnecessary as well now that you use `AddRange(...)`.